### ISSUE TITLE
Append an array

### ISSUE THREAD

**Issue by jroper** [typesafehub/config#227]
Typesafe config currently supports appending an item to an array, but not appending/concatenating an array to an array. Perhaps this could be done with `++=` syntax.

Use case is described [here](https://groups.google.com/d/msg/play-framework-dev/9IZOhzQ8e9Y/rgR5_L0c79IJ).

**Comment by havocp** [comment]
It is supported already, just put two arrays next to each other on the same line. There's no operator for it just adjacency. For `foo ++= [1,2]` you would write `foo = ${foo} [1,2]` at the moment.

**Comment by julienrf** [comment]
Yes, that’s [what I did](https://github.com/julienrf/pfe-samples/blob/7a341dd985c118dacc02a77093bea1799f73570d/pfe-scala/conf/application.conf#L23). I was just looking for a more convenient way to write it.

**Comment by fommil** [comment]
my usecase seems slightly different to the above. In scalafmt's config I'd like to be able to add things to the default array. `foo ++= [my extras]` would be a really great thing to be able to do.

**Comment by raboof** [comment]
@fommil do you have an example of a scalafmt option that has this problem? I suspect scalafmt might be doing something like `effectiveValue = valuesFromConfig().getOrElse(defaultValues())`, which would make this a scalafmt problem rather than a problem with this library - but I couldn't find a good example having a quick look through the code.

**Comment by fommil** [comment]
pinging @olafurpg

**Comment by dwinters-gpsw** [comment]
I see where Typesafe config supports appending an array to an array when one of the arrays is a literal such as this:

```
  foo = ${foo} [1,2]
```

However, I'm not able to concatenate two arrays together by using two substitutions for the arrays:

```
  a : [ 1, 2 ]
  b : [ 3, 4 ]
  c : ${a} ${b}
```

I get the following exception:

com.typesafe.config.ConfigException$WrongType: foo.conf: 60: Cannot concatenate object or list with a non-object-or-list, SimpleConfigList([1,2]) and ConfigString(" ") are not compatible

Is there not a way to concatenate two arrays by using substitutions?

**Comment by dwinters-gpsw** [comment]
I see that the issue with concatenating two arrays by substitution is covered in the following issue:

https://github.com/lightbend/config/issues/370

There's also a nice workaround included in that issue.
